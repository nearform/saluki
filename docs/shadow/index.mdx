import { Shadow } from './'

# Shadow

---

## Example

<Shadow />

```js
const Box = styled.h1`
  ${size.height.medium}
  ${background.black}
  ${shadow.large}
`
```

## Rules

```js
shadow.small
shadow.medium
shadow.large
```

## Configuration

Configurable via;

```
shadow
```

## Defaults

```js
{
  small: '0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24)',
  medium: '0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23)',
  large: '0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22)'
}
```
