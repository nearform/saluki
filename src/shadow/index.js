export const defaultRules = {
  small: '0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24)',
  medium: '0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23)',
  large: '0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22)'
}

function generateRules(rules) {
  const newRules = {}

  for (const [key, value] of Object.entries(rules)) {
    newRules[key] = {
      'box-shadow': value
    }
  }

  return newRules
}

export default function(shadow) {
  return generateRules(shadow)
}
