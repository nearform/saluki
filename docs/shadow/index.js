import styled from 'styled-components/macro'
import { background, shadow, size } from '../../src'

export const Shadow = styled.h1`
${size.height.medium}
  ${background.color.black}
  ${shadow.large}
`
