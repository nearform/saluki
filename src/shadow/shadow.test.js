import shadow, { defaultRules } from './'

describe('shadow', () => {
  describe('when no custom config is provided', () => {
    it('should return the default config', () => {
      expect(shadow(defaultRules)).toMatchSnapshot()
    })
  })

  describe('when custom config is provided', () => {
    describe('when the custom config overrides a value', () => {
      it('should replace the default value', () => {
        const combinedRules = {
          ...defaultRules,
          ...{ large: '5px 10px rgba(0,0,0,0.25)' }
        }
        expect(shadow(combinedRules)).toMatchSnapshot()
      })
    })

    describe('when the custom config provides a new value', () => {
      it('should add the value', () => {
        const combinedRules = {
          ...defaultRules,
          ...{ solid: '10px 20px' }
        }
        expect(shadow(combinedRules)).toMatchSnapshot()
      })
    })
  })
})
