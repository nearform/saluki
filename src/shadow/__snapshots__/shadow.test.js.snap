// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shadow when custom config is provided when the custom config overrides a value should replace the default value 1`] = `
Object {
  "large": Object {
    "box-shadow": "5px 10px rgba(0,0,0,0.25)",
  },
  "medium": Object {
    "box-shadow": "0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23)",
  },
  "small": Object {
    "box-shadow": "0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24)",
  },
}
`;

exports[`shadow when custom config is provided when the custom config provides a new value should add the value 1`] = `
Object {
  "large": Object {
    "box-shadow": "0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22)",
  },
  "medium": Object {
    "box-shadow": "0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23)",
  },
  "small": Object {
    "box-shadow": "0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24)",
  },
  "solid": Object {
    "box-shadow": "10px 20px",
  },
}
`;

exports[`shadow when no custom config is provided should return the default config 1`] = `
Object {
  "large": Object {
    "box-shadow": "0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22)",
  },
  "medium": Object {
    "box-shadow": "0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23)",
  },
  "small": Object {
    "box-shadow": "0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24)",
  },
}
`;
